### PR TITLE
refactor: Code refactor

### PR DESCRIPTION
**Pull request by JackTan25**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
ReplacementLogEntry is same with before, if we make it as a enum, there will be many places need to process
unreachable branch (like recluster and so on, anywhere use it)
Summary about this PR

- Closes #issue
- reference to advice in #11937

### CODE REVIEW

**Review by zhyass — APPROVED** [review]
LGTM

**Comment by BohuTANG** [comment]
askbend:summary

File: src/query/storages/fuse/src/operations/mutation/mutation_part.rs [hunk 1]
```diff
@@ -24,6 +23,48 @@ use storages_common_table_meta::meta::ClusterStatistics;
 
 use crate::pruning::DeletedSegmentInfo;
 
+#[derive(serde::Serialize, serde::Deserialize, PartialEq)]
+pub enum MutationEnum {
```
Comment (dantengsky) [inline src/query/storages/fuse/src/operations/mutation/mutation_part.rs:27]: how about naming it `Mutation`?

File: src/query/storages/fuse/src/pruning/fuse_pruner.rs [hunk 2]
```diff
@@ -288,28 +288,40 @@ impl FusePruner {
...
-                                        compact_segment_info.summary.clone(),
-                                    ),
-                                })
-                            } else {
-                                res.extend(
-                                    block_pruner
-                                        .pruning(segment_location.clone(), compact_segment_info)
-                                        .await?,
-                                );
+                            // for delete_prun
```
[the remark this answers is not in the document]
Comment (JackTan25) [reply src/query/storages/fuse/src/pruning/fuse_pruner.rs:294]: yes

File: src/query/storages/fuse/src/operations/mutation/mutation_part.rs (hunk 1 above)
Comment (JackTan25) [reply src/query/storages/fuse/src/operations/mutation/mutation_part.rs:27]: ok

**Review by dantengsky — APPROVED** [review]

**Pull request merged**